Added capability to RasterImage layer to read geogeaphic info from .aux.xml file

ma15569 opened the pull request:
See mail https://sourceforge.net/p/jump-pilot/mailman/message/37833644/ for more details

edeso commented on src/org/openjump/core/ccordsys/utils/ProjUtils.java:
```diff
@@ -486,6 +487,7 @@ public static SRSInfo getSRSInfoFromLayerSource(RasterImageLayer layer)
         // fallthrough
         if (srsInfo == null) {
             srsInfo = ProjUtils.getSRSInfoFromAuxiliaryFile(fileSourcePath);
+             srsInfo.setSource(AUXFILE_SRS);
```
this should be set in getSRSInfoFromAuxiliaryFile() **and only if** the ".aux.xml" file is indeed available and the srs is read from it.

ma15569 replied:
I will remove this line as I didn't realize that ProjUtils.getSRSInfoFromAuxiliaryFile() has already a setSource that points to the path of aux file (strangely it fails sometimes, I will inverstigate later)

edeso commented on src/org/openjump/core/rasterimage/AddRasterImageLayerWizard.java:
```diff
@@ -263,175 +263,114 @@ private void addImage(WorkbenchContext context, Envelope envelope,
-                    .getFrame()
-                    .warnUser(
-                            I18N.getInstance().get("org.openjump.core.rasterimage.AddRasterImageLayerWizard.can-not-determine-image-dimensions"));
-            return null;
```
this is wrong. you added tabs to the indentation (spaces only before) and did some unneeded reformatting. i cannot see clearly what you changed here.

please fix the whole file. compare with previous version from branch main.

ma15569 replied:
I see the tabs+spaces. I will correct it

Should I close this pull request and open a new one?

edeso replied:
no. you can just push to the branch this PR is based on :)

jratike80 commented:
I believe (not sure) that aux.xml files may exist even without georeferencing info. Does the code skip aux.xml in such case and search for world files etc.?

edeso commented:
hey Jukka,

it's pretty convoluted by now. but yeah, in theory it should try all methods until one succeeds. should be this part. try geotiff or if not avail, try sidecar file.

https://github.com/openjump-gis/openjump/blob/main/src/org/openjump/core/ccordsys/utils/ProjUtils.java#L468-L499

ma15569 commented on src/language/jump.properties:
```diff
@@ -2964,6 +2964,7 @@ ui.zoom.ZoomToCoordinatePlugIn.enter-two-values=Enter two values!
 ui.zoom.ZoomToCoordinatePlugIn.{0}-is-not-a-number=''{0}'' is not a number!
 org.openjump.core.ccordsys.No-srs=No SRS info recorded from data source
 org.openjump.core.ccordsys.Embedded-srs=SRS info is included in data source
+org.openjump.core.ccordsys.AuxFile-srs=SRS info is included into an auxiliary file
```
This codes will be removed. See comment in [src/org/openjump/core/ccordsys/utils/ProjUtils.java](https://github.com/openjump-gis/openjump/pull/97/files/df818342b30c031c9c337605d8324c2c20eeb07a#diff-ba250e9f8be116c2fc02528c7b1b08e2b64faba706807d1d8f351bbe512a4573)

ma15569 commented:
Hi Ede,
this request seems still in progress.

ma15569 commented:
😕

edeso commented:
thx for the reminder. just ravelling.

will try to fit in a review while riding the train Mo or Tue :)

edeso commented on src/language/jump_es.properties:
```diff
@@ -3008,3 +3008,4 @@ org.openjump.core.ui.plugin.tools.MergeSelectedPolygonsWithNeighbourPlugIn.merge
 org.openjump.core.ui.plugin.tools.MergeSelectedPolygonsWithNeighbourPlugIn.minimum-area=#T:minimum area
 org.openjump.core.ui.plugin.tools.MergeSelectedPolygonsWithNeighbourPlugIn.Skip-null-values=#T:Skip features with null attribute value
 jump.features-processed=elementos procesados
+org.openjump.core.ccordsys.AuxFile-srs=SRS se incluye en un archivo auxiliar
```
this new key needs to be added to **all language files**
especially to the english version `src/language/jump.properties` with english translation

edeso commented on src/org/openjump/core/rasterimage/AddRasterImageLayerWizard.java:
```diff
@@ -255,6 +260,12 @@ private void addImage(WorkbenchContext context, Envelope envelope,
      * WorldFileHandler.readWorldFile()
      * 
+     * [Giuseppe Aruta - 04/27/2023]. The method has been "slimed" from the previous version:<br>
+     * <li>a) The first part that build the Envelope from image info has been removed and delegated
```
do you mean "slimmed down" meaning reduced in complexity/size?

edeso reviewed (changes requested):
just two small issues, the files being the bigger one. getting there :)

ma15569 commented:
Hi Ede,

a) I removed the code
 _org.openjump.core.ccordsys.AuxFile-srs=SRS se incluye en un archivo auxiliar as_ 
from the Spanis lang file as it is not used anymore
Meanwhile I added new translations to that file

b) AdddRasterIageLayerWizard.  I correct the typo from "slimed"  to "optimized".

Not sure if the corrections are visible on main brench

Peppe

ma15569 commented:
??

edeso commented:
this line in _ProjUtils_
```
private static final String AUXFILE_SRS = I18N.getInstance().get("org.openjump.core.ccordsys.AuxFile-srs");
```
is obsolete now, is it not?

edeso commented:
is this still ongoing or was it resolved inbetween?

ma15569 commented:
Hi Ede, I prefer to abandon this request and focus on developing my own external plugins, where I have more possibilities to manage the code. This option has however been added to my Raster Tools. which have a parallel system of importing raster data. If you don't mind, I will close the request.

edeso commented:
>  If you don't mind, I will close the request.

sure. go ahead :)
